### ISSUE TITLE
btsieve: `Confirmations Needed` has been removed

### ISSUE THREAD

**Issue by bonomat** [comit-network/comit-rs#591]
Currently the LQS allows to create a query with a field `confirmations_needed`. This is used to give more certainty over a transaction. However, this feature is poorly implemented:

LQS watches new blocks and filters the transaction according the given queries. If a suitable transaction was found, the query _matches_ and the transaction will be added to a repository. Depending on the field `confirmations_needed` the transaction is either added to a _pending_ repository (`nr_of_confirmation<confirmations_needed`) or to the result repository ((`nr_of_confirmation>confirmations_needed`). 

For each new incoming block, the transactions in the _pending_ repository get updated, i.e. the amount of `confirmations_needed` gets decremented. However, with this approach we would not be able to detect if a fork has happened, and this transaction was actually never included in the blockchain. 

Hence, we would need to check for every transaction (for a query with `confirmations_needed >0`) with the blockchain whether this transaction is still included and if the amount of required confirmation has been fulfilled.

Once done, open issue to make comit_node uses it.

**Comment by bonomat** [comment]
Moving this back into New Issues as the description is really outdated.

**Comment by thomaseizinger** [comment]
New issue to be created once btsieve overhaul is done.
